pil_image_to_array_block.py

```diff
@@ -0,0 +1,16 @@
+import numpy as np
+
+from nio.block.base import Block
+from nio.properties import VersionProperty
+
+
+class PILImageToArray(Block):
+
+    """ Convert a PIL Image to a numpy array """
+
+    version = VersionProperty('0.1.0')
+
+    def process_signals(self, signals, input_id='default'):
+        for signal in signals:
+            signal.image = np.array(signal.image)
```

Comment:
This should typically use an expression, and preferably the enrich signals mixin
```python
image_expr = Property("PIL Image", default="{{ $image }}")
```
```python
out_signals = []
for signal in signals:
    out_signals.append(self.get_output_signal({
        'image': np.array(self.image_expr(signal))
    }, signal))
self.notify_signals(out_signals)
```

We could probably clean up the enrich signals interface to be a bit friendlier for lists of signals, but using it will allow you to configure whether you want it to append, overwrite, or replace the incoming signal with the array data. You will thank me later for having this ability from the service :)

Reply:
Agreed! I only did it this way to be consistent with the other blocks in this repo. A refactor is in order, both of these suggestions should be implemented.